### ISSUE TITLE
_cdist_backward ignores p and always computes the p=2 gradient

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5023]
### Describe the bug

`_cdist_backward` only computes the correct gradient for `p = 2`. The kernel takes `p` as an argument but never reads it — the math is hardcoded to the Euclidean case. The op is registered unconditionally in `src/flag_gems/__init__.py:78`, so any `p != 2` silently returns a wrong gradient with no warning or error.

In `src/flag_gems/ops/cdist_backward.py:100`:

```python
contrib = grad[:, None] * diff / (cdist[:, None] + eps)
```

That is `grad * diff / dist`, which is the `p = 2` gradient. The general form is `grad * sign(diff) * |diff|^(p-1) / dist^(p-1)`, with separate cases for `p = 0`, `p = 1`, and `p = inf`. `p` is passed into the kernel signature but is unused in the body.

The test never catches this because `tests/test_cdist_backward.py:44` hardcodes `p = 2.0` and there is no parametrization over `p`.

The kernel was added in #3507.

### Reproduction

```python
import torch, flag_gems

torch.manual_seed(0)
b, n1, n2, d = 2, 8, 5, 4
x1 = torch.randn(b, n1, d, device="cuda")
x2 = torch.randn(b, n2, d, device="cuda")
g = torch.randn(b, n1, n2, device="cuda")

for p in [1.0, 2.0, 3.0, float("inf")]:
    cd = torch.cdist(x1, x2, p=p)
    ref = torch.ops.aten._cdist_backward(g, x1, x2, p, cd)
    with flag_gems.use_gems():
        res = torch.ops.aten._cdist_backward(g, x1, x2, p, cd)
    print(f"p={p:<5} max_abs_err={(res - ref).abs().max().item():.6f}")
```

Output (NVIDIA H100, float32):

```
p=1.0   max_abs_err=3.929809
p=2.0   max_abs_err=0.000000
p=3.0   max_abs_err=0.957097
p=inf   max_abs_err=2.918434
```
